source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -426,6 +433,129 @@ int StreamHandleWrapper::luaLogCritical(lua_State* state) {
   return 0;
 }
 
+const EVP_MD* StreamHandleWrapper::getDigest(const absl::string_view& name) {
+  std::string hash = absl::AsciiStrToLower(name);
+
+  // Hash Hash algorithms set refers
+  // https://github.com/google/boringssl/blob/ff62b38b4b5a0e7926034b5f93d0c276e55b571d/include/openssl/digest.h
+  if (hash == "md4") {
+    return EVP_md4();
+  } else if (hash == "md5") {
+    return EVP_md5();
+  } else if (hash == "sha1") {
+    return EVP_sha1();
+  } else if (hash == "sha224") {
+    return EVP_sha224();
+  } else if (hash == "sha256") {
+    return EVP_sha256();
+  } else if (hash == "sha384") {
+    return EVP_sha384();
+  } else if (hash == "sha512") {
+    return EVP_sha512();
+  } else if (hash == "md5_sha1") {
+    return EVP_md5_sha1();
+  } else {
+    return nullptr;
+  }
+}
+
+int StreamHandleWrapper::luaVerifySignature(lua_State* state) {
+  // Step 1: get public key
+  auto ptr = lua_touserdata(state, 2);
+  auto key = reinterpret_cast<EVP_PKEY*>(ptr);
+
+  // Step 2: initialize EVP_MD_CTX
+  EVP_MD_CTX* ctx = EVP_MD_CTX_new();
```

Comment:
what if ctx is NULL?

Reply:
It will not trigger any segment fault due to the implementation of both `EVP_MD_CTX_free` and `EVP_DigestVerifyInit` support passing NULL as a parameter. 

`EVP_MD_CTX_free` will return if the parameter is NULL. 
`EVP_DigestVerifyInit` will try to call `EVP_MD_CTX_new` again if the parameter is NULL. If the context is still NULL, the function will return 0.